File: NQueens/nqueens.py
def printSolution(board):
    for i in range(len(board)):
        for j in range(len(board)):
            if board[i][j] == 1:
                print("Q",end=" ")
            else:
                print(".",end=" ")
        print()

def isSafe(board, row, col):

    for i in range(0, col):
        if (board[row][i] == 1):
            return False
        
    for i, j in zip(range(row, -1, -1),
                    range(col, -1, -1)):        
        if (board[i][j] == 1):
            return False
        
    for i, j in zip(range(row, len(board), 1),
                    range(col, -1, -1)):         
         if (board[i][j] == 1):
            return False
    
    return True
        
def solveNQ(board, col):
    if col >= len(board):
        return True

    for i in range(0, len(board)):
        if (isSafe(board, i, col)):
            board[i][col] = 1

            if solveNQ(board, col +1) == True:
                return True
            
            board[i][col] = 0

    return False

def main():
    board = [[0, 0, 0, 0, 0, 0, 0, 0],
             [0, 0, 0, 0, 0, 0, 0, 0],
             [0, 0, 0, 0, 0, 0, 0, 0],
             [0, 0, 0, 0, 0, 0, 0, 0],
             [0, 0, 0, 0, 0, 0, 0, 0],
             [0, 0, 0, 0, 0, 0, 0, 0],
             [0, 0, 0, 0, 0, 0, 0, 0],
             [0, 0, 0, 0, 0, 0, 0, 0],
             ]
 
    if solveNQ(board, 0) == False:
        print("Solution does not exist")
        return False
 
    printSolution(board)

if __name__ == "__main__":
    main()

File: NQueens/nqueens.c
#include <stdbool.h>
#include <stdio.h>

#define N 4

void printSolution(int board[N][N]){
    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            if(board[i][j])
                printf("Q ");
            else
                printf(". ");
        }
        printf("\n");
    }
}

bool isSafe(int board[N][N], int row, int col){ 
    for (int i = 0; i < col; i++)
        if (board[row][i])
            return false;

    for (int i = row, j = col; i >= 0 && j >= 0; i--, j--)
        if (board[i][j])
            return false;

    for (int i = row, j = col; j >= 0 && i < N; i++, j--)
        if (board[i][j])
            return false;
 
    return true;
}
 

bool solveNQ(int board[N][N], int col){

    if (col >= N){
        return true;
    }
        
    for (int i = 0; i < N; i++) {
         
        if (isSafe(board, i, col)) {

            board[i][col] = 1;
 
            if (solveNQ(board, col + 1))
                return true;

            board[i][col] = 0;
        }
    }
 
    return false;
}



int main(int argc, char *argv[]) {
    int board[N][N] = { { 0, 0, 0, 0 },
                        { 0, 0, 0, 0 },
                        { 0, 0, 0, 0 },
                        { 0, 0, 0, 0 } };
 
    if (solveNQ(board, 0) == false) {
        printf("Solution does not exist");
    } else {
        printSolution(board);
    }   

    return 0;
}